src/hnsw/build.c

```diff
@@ -531,3 +541,75 @@ void ldb_ambuildunlogged(Relation index)
     // todo::
     elog(ERROR, "hnsw index on unlogged tables is currently not supported");
 }
+
+void ldb_reindex_external_index(Oid indrelid)
+{
+    BlockNumber          HEADER_BLOCK = 0;
+    Relation             index_rel;
+    Buffer               buf;
+    Page                 page;
+    char                *metric_kind;
+    char                *index_name;
+    HnswIndexHeaderPage *headerp;
+    FmgrInfo             reindex_finfo = {0};
+    char    *function_sig = "_lantern_reindex_external_index(text, text, integer, integer, integer, integer)";
+    Oid      function_oid;
+    uint32_t dim = 0;
+    uint32_t m = 0;
+    uint32_t ef_construction = 0;
+    uint32_t ef = 0;
+
+    PG_TRY();
+    {
+        function_oid = DatumGetObjectId(DirectFunctionCall1(regprocedurein, CStringGetDatum(function_sig)));
```

Comment:
Is there a reason to prefer the tr/catch approach? 
There are more granular function lookup APIs that will not need exceptions.
(I just have not seen the try/catch machinery be used for this inside postgres codebase, if it is, then probably fine to leave it as is, since this is probably shorter)